src/App.js

```diff
@@ -2,44 +2,38 @@ import React, { Component } from "react";
 import TopSection from "./components/TopSection";
 import MovieCardContainer from "./components/MovieCardContainer";
 import SingleMovie from "./components/SingleMovie";
-import apiCalls from "./apiCalls"
+import { Route } from "react-router-dom";
+import apiCalls from "./apiCalls";
 import "./App.css";
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
       movies: [],
-      movie: [],
       error:""
     };
   }
 
   componentDidMount() {
-    Promise.all([apiCalls.getMovies()])
-    .then(data => this.setState({movies: data[0].movies,error:""}))
-    .catch(error => this.setState({error:error.message}))
+    apiCalls.getMovies()
+      .then(data => this.setState({movies: data.movies,error:""}))
+      .catch(error => this.setState({error:error.message}))
   }
 
-  goHome = () => {
-    this.setState({movie:[]})
-  }
-
-  selectMovie = (id) => {
-    Promise.all([apiCalls.getMovies(id)])
-    .then(data => this.setState({movie: [data[0].movie],error:""}))
-    .catch(error => this.setState({error:error.message}))
-  }
-
-
   render() {
     return (
       <main className="App">
-        <TopSection goHome={this.goHome} length={this.state.movie.length}/>
-        {this.state.error && <h2>An error has occured, please try your request again later.</h2>}
-        {!this.state.movie.length ? <MovieCardContainer selectMovie={this.selectMovie} movies={this.state.movies}/> : <SingleMovie movie={this.state.movie[0]}/>}
+        <TopSection />
+        <Route exact path="/" render={() => {
+           return <MovieCardContainer movies={this.state.movies}/>
+        }}
+        />
+        <Route exact path="/:movie_id" render={ ({ match }) => {
+          return (<SingleMovie movieId={parseInt(match.params.movie_id)}/>)
+        }}
+        />
       </main>
-
```

Comment:
We should consider using a switch in app to handle missing movies, and make a Project issue for this